Update missing/broken URL's in the documentation

hchandad opened the pull request:
Related issue: #386 

Working on updating missing/broken URL's in the documentation.

This PR updates affects the following pages:
- [/blog/2023-10-22-unikraft-releases-pandora](http://localhost:3000/blog/2023-10-22-unikraft-releases-pandora)
- [/blog/2017-12-05-unleashing-unikernels](http://localhost:3000/blog/2017-12-05-unleashing-unikernels)
- [/docs/getting-started/integrations/hcp-packer](http://localhost:3000/docs/getting-started/integrations/hcp-packer)
- [/docs/internals/architecture](http://localhost:3000/docs/internals/architecture)
- [/docs/internals/booting](http://localhost:3000/docs/internals/booting)
- [/docs/getting-started/integrations/github-actions](http://localhost:3000/docs/getting-started/integrations/github-actions)
- [/docs/internals/writing-tests](http://localhost:3000/docs/internals/writing-tests)
- [/docs/internals/architecture](http://localhost:3000/docs/internals/architecture)
- [/guides/catalog-using-firecracker](http://localhost:3000/guides/catalog-using-firecracker)
- [/docs/cli/running](http://localhost:3000/docs/cli/running)
- [/docs/internals/build-system](http://localhost:3000/docs/internals/build-system)
- [/docs/contributing/unikraft](http://localhost:3000/docs/contributing/unikraft)
- [/docs/contributing/coding-conventions](http://localhost:3000/docs/contributing/coding-conventions)

It also updates the internals/cli sidebar

razvand commented:
Thanks so much for your work, @hchandad . Please start commit messages with capital letters. Follow the instructions [here](https://github.com/unikraft/docs/blob/main/content/docs/contributing/docs.mdx) related to contributing.

Please check [this](https://unikraft.org/docs/contributing/unikraft#commits) related to doing commit messages.

RaduNichita commented:
Hey,

Just a qq: what do you think about having one single commit for all the fixes? I don't feel like we need to have so many commits for fixing some links?

@razvand  any opinion?

StefanJum commented:
> Just a qq: what do you think about having one single commit for all the fixes? I don't feel like we need to have so many commits for fixing some links?

Agree with this, but only on short fixes like urls. New (old) content that you added or anything else more than a couple of lines can stay in a separate commit.

razvand commented:
> > Just a qq: what do you think about having one single commit for all the fixes? I don't feel like we need to have so many commits for fixing some links?
> 
> Agree with this, but only on short fixes like urls. New (old) content that you added or anything else more than a couple of lines can stay in a separate commit.

Agree with @StefanJum . @hchandad , sorry for bothering you again with this. Could you please create a single commit with all URL fixes? You can do separate commits for non-URL-related fixes.

hchandad commented:
@RaduNichita @StefanJum @razvand 

- I grouped the single line changes, and moving the assets to the public folder in commit `a4b8101`
- The included sections from the old documentation are in separate commits, and i updated `next-redirect.js` with redirects for each to keep old references working.
- I added a new section in docs/contributing/docs.mdx to describe how to use linkcheckers as part of testing, in `d41a00b`

razvand reviewed (commented):
Hi, @hchandad . This looks good. Please add the `Signed-off-by` line at the end of all message lines in a commit.

StefanJum commented:
Looks great @hchandad, thank you. Besides what @razvand said, please add a very short output example for the `linkcheck` section, similar to what we have for the other linters. All good besides that.

hchandad commented:
@razvand @StefanJum Done

razvand reviewed (approved):
Thanks, @hchandad . This looks good to me.

Reviewed-by: Razvan Deaconescu <razvand@unikraft.io>

StefanJum reviewed (approved):
Thanks @hchandad 

Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>
Approved-by: Stefan Jumarea <stefanjumarea02@gmail.com>
